File: model_dflowfm_blackbox/tests/estuary_kalman_FMSuite2019.01/plot_sequential.m
% Sequential simulation:
% Make plot to show difference between observations and predicted values
%
[t, obs, pred_f pred_a]=load_sequential('SequentialSimulation_results');

gcf=1; figure(gcf);
plot(pred_f(1,1:length(t)));
hold on
plot(obs(1,1:length(t)),'k');
title('Obs01');
legend('prediction','observation');
xlabel('time');

gcf = gcf + 1; figure(gcf);
plot(pred_f(2,1:length(t)));
hold on
plot(obs(2,1:length(t)),'k');
title('Obs02');
legend('prediction','observation');
xlabel('time');

gcf = gcf + 1; figure(gcf);
plot(pred_f(3,1:length(t)));
hold on
plot(obs(3,1:length(t)),'k');
title('Obs03');
legend('prediction','observation');
xlabel('time');

diff=pred_f-obs;
gcf = gcf + 1; figure(gcf);
plot(diff')
title('Difference between model and observations');
ylabel('prediction-observation');
xlabel('time');

[nobs,ntimes]=size(diff)
for iobs=1:nobs
   rms=norm(diff(iobs,:))/sqrt(ntimes);
   disp(['RMS of location ',num2str(iobs),' ',num2str(rms)]);
end

File: model_dflowfm_blackbox/tests/estuary_kalman_pli-file/plot_sequential.py
#! /usr/bin/python
# -*- coding: utf-8 -*-
"""
   Sequential simulation:
   Make plot to show difference between observations and predicted values
"""

import SequentialSimulation_results as seqsim
import matplotlib.pyplot as plt

nobs = len(seqsim.obs)
obs = seqsim.obs
pred_f = seqsim.pred_f_central

fig1 = plt.figure()
plt.plot(pred_f[:,0])
plt.plot(obs[:,0])
plt.title('Obs01')
plt.xlabel('time')
plt.legend(['prediction','observation'])

fig2 = plt.figure()
plt.plot(pred_f[:,1])
plt.plot(obs[:,1])
plt.title('Obs02')
plt.xlabel('time')
plt.legend(['prediction','observation'])

fig3 = plt.figure()
plt.plot(pred_f[:,2])
plt.plot(obs[:,2])
plt.title('Obs03')
plt.xlabel('time')
plt.legend(['prediction','observation'])

diff = pred_f - obs
fig4 = plt.figure()
plt.plot(diff)
plt.title('Difference between model and observations')
plt.ylabel('prediction-observation')
plt.xlabel('time')

plt.show()


